### PR TITLE
Fix: Square atom_norm in non-Huber energy and virial loss calculations.

### PR DESCRIPTION
**Pull request by anyangml**
This is to normalize all L2 loss so that in multitask training, the scale of gradients becomes independent of system size. Do we want to update pref in examples? @iProzd @njzjz 

## Summary by CodeRabbit

* **New Features**
  * Added an "intensive" option to energy/virial loss configs to toggle alternative size normalization (legacy 1/N vs intensive 1/N²) for non-Huber MSE loss across energy and spin loss implementations and config helpers.

* **Bug Fixes**
  * Serialization updated to persist the new option and remain backward-compatible when loading older configs.

* **Documentation**
  * Training docs and examples updated; fixed "magnetic" typo and added math and guidance for the `intensive` option.

* **Tests**
  * New and extended tests validate intensive vs legacy scaling across scenarios and backends.

### CODE REVIEW

**Comment by iProzd** [comment]
Some AI suggestions are needed to be resovled. Others LGTM.

File: deepmd/pt/loss/ener.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def forward(
                         l2_ener_loss.detach(), find_energy
                     )
                 if not self.use_huber:
-                    loss += atom_norm * (pref_e * l2_ener_loss)
+                    loss += atom_norm**2 * (pref_e * l2_ener_loss)
                 else:
```
[the remark this answers is not in the document]
Comment (anyangml) [reply deepmd/pt/loss/ener.py:267]: @copilot apply changes based on [this feedback](https://github.com/deepmodeling/deepmd-kit/pull/5332#discussion_r2972779846)

File: deepmd/tf/loss/ener.py [hunk 2]
```diff
@@ -370,7 +370,9 @@ def build(
             )
         if self.has_v:
             if not self.use_huber:
-                loss += global_cvt_2_ener_float(atom_norm * (pref_v * l2_virial_loss))
+                loss += global_cvt_2_ener_float(
+                    atom_norm**2 * (pref_v * l2_virial_loss)
+                )
```
[the remark this answers is not in the document]
Comment (anyangml) [reply deepmd/tf/loss/ener.py:399]: @copilot apply changes based on [this feedback](https://github.com/deepmodeling/deepmd-kit/pull/5332#discussion_r2972779856)

**Review by iProzd — APPROVED** [review]

File: deepmd/dpmodel/loss/ener.py [hunk 3]
```diff
@@ -90,6 +90,14 @@ class EnergyLoss(Loss):
...
         Instead of computing loss on force components, computes loss on ||F_pred - F_label||_2.
         This treats the force vector as a whole rather than three independent components.
+    intensive : bool
+        If true, the non-Huber MSE energy and virial losses use intensive normalization,
+        i.e. a 1/N^2 factor instead of the legacy 1/N scaling. This matches per-atom
+        RMSE-style normalization for those terms. MAE and Huber modes use different
+        scaling and are not affected in the same way by this flag.
+        If false (default), the legacy normalization is used for the affected terms.
+        The default is false for backward compatibility with models trained using
+        deepmd-kit <= 3.0.1.
```
Comment (njzjz) [inline deepmd/dpmodel/loss/ener.py:100]: what does 3.0.1 mean? It's not the current version.

**Comment by njzjz** [comment]
@anyangml The code where I commented was not changed?

image

File: deepmd/dpmodel/loss/ener_spin.py [hunk 4]
```diff
@@ -50,6 +50,14 @@ class EnergySpinLoss(Loss):
...
     loss_func : str
         Loss function type: 'mse' or 'mae'.
+    intensive : bool
+        If true, the MSE energy and virial terms use intensive normalization,
+        i.e. an additional normalization by the square of the number of atoms
+        (1/N^2) instead of the legacy (1/N) behavior. This keeps those MSE loss
+        terms consistent with per-atom RMSE reporting and less dependent on
+        system size. This option does not change the MAE formulation, which is
+        handled separately. The default is false for backward compatibility with
+        models trained using deepmd-kit <= 3.1.3.
```
Comment (wanghan-iapcm) [inline deepmd/dpmodel/loss/ener_spin.py]: intensive -> intensive_ener_virial 
to be more self explanatory.

**Review by wanghan-iapcm — APPROVED** [review]

**Review by njzjz — APPROVED** [review]
